presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -1404,6 +1413,28 @@ void PrestoServer::reportNodeStatus(proxygen::ResponseHandler* downstream) {
   http::sendOkResponse(downstream, json(fetchNodeStatus()));
 }
 
+void PrestoServer::handleGracefulShutdown(const std::vector<std::unique_ptr<folly::IOBuf>>& body, proxygen::ResponseHandler* downstream){
+    if (body.size()==1 && body[0]->moveToFbString() == "\"SHUTTING_DOWN\"") {
+         // Print message and initiate shutdown
```

Comment:
We probably don't need all these comments as they explains what happens which is apparent from the code. We use the Velox comment guidelines: https://github.com/facebookincubator/velox/blob/main/CONTRIBUTING.md#coding-best-practices (see "Code Comments" section).

Reply:
I agree.